### ISSUE TITLE
Introduce a custom `ApolloResponse.toString()` 

### ISSUE THREAD

**Issue by rohandhruva** [apollographql/apollo-kotlin#5872]
### Use case

Since `ApolloResponse` changed to being a standard `class` (I think it used to be a `data class`?), the `toString()` now is not very useful because it prints the default hashcode / instance code based `toString`. 

Can we introduce a custom `toString` for `ApolloResponse`? While printing all of `data` does not make sense, I'm trying to think what might be useful -- perhaps the cache key?

**Comment by martinbonnin** [comment]
Agreed. This is also recommended by the [Kotlin API guidelines](https://kotlinlang.org/docs/jvm-api-guidelines-debuggability.html#always-provide-a-tostring-method). While I wouldn't put `toString()` on each and every class, `ApolloResponse` is central enough that it deserves one.

**Comment by martinbonnin** [comment]
Available in `4.2.0`
